android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -1413,6 +1413,34 @@ public void setPreventsDisplaySleepDuringVideoPlayback(boolean preventsDisplaySl
 
     public void setSelectedTrack(int trackType, String type, Dynamic value) {
         if (player == null) return;
+        if (!TextUtils.isEmpty(type) && type.equals("maxResolution")) {
+            int maxBitRate = 0;
+            int foundHeight = 0;
+            WritableArray tracks = getVideoTrackInfo();
```

Comment:
this code is not efficient check the code inside getVideoTrackInfo and compare it with  setSelectedTrack the part after getCurrentMappedTrackInfo

you will find they are very similar 

my main issue is you are dealing with maps and you dont have to, you can just get the tracks directly without conversion